browser/brave_wallet/brave_wallet_provider_impl_unittest.cc

```diff
@@ -273,7 +274,7 @@ class BraveWalletProviderImplUnitTest : public testing::Test {
   void Navigate(const GURL& url) { web_contents()->NavigateAndCommit(url); }
 
   GURL GetOrigin() {
-    return web_contents()->GetMainFrame()->GetLastCommittedURL().GetOrigin();
+    return web_contents()->GetMainFrame()->GetLastCommittedOrigin().GetURL();
```

Comment:
Maybe we can change the method signature as well?

Reply:
In what way? The return value is used in calling `EthJsonRpcController::NotifySwitchChainRequestProcessed`, which takes `const GURL& origin`.

Reply:
I gather it should be either `GURL GetURL()` or `Origin GetOrigin()`. But we can leave it alone if there are too much changes

Reply:
I think `Origin GetOrigin()` would make more sense to me here if we also moved the `.GetURL()` part to the call sites. Otherwise maybe `GetOriginAsURL()`?

Reply:
I actually looked for a while into changing the signature not only of this one, but also of many other methods, to use an url::Origin instead of a GURL when actually referring to "origins", and the change started spreading way too much over the codebase (e.g. mojo interfaces) that was clearly out of scope for the rebasing work, I think.

Maybe we should file a bug, though, to "audit and adapt Brave's code base to stop using GURLs and std::strings for things that are meant to be origins, and do that separate from this rebasing work. WDYT?

Reply:
filing a bug sounds good to me!

Reply:
Filed in https://github.com/brave/brave-browser/issues/19849